ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/authorization/auth/AuthServiceStaticMap.scala

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.server.api.authorization.auth
+
+import java.util.concurrent.{CompletableFuture, CompletionStage}
+
+import com.daml.ledger.participant.state.v1.{AuthService, Claims}
+import io.grpc.Metadata
+
+/** An AuthService that matches the value of the `Authorization` HTTP header against
+  * a static map of header values to [[Claims]].
+  *
+  * Note: This AuthService is meant to be used for testing purposes only.
+  */
+class AuthServiceStaticMap(claims: Map[String, Claims]) extends AuthService {
```

Comment:
Consider turning this into a `PartialFunction[String, Claims]` to that something like this can be passed in without hard-coding anything specific to the ledger setup:

```
new AuthServiceStaticMap({
  case "root" => Claims.wildcard
  case "admin" => Claims(Seq(ClaimAdmin))
  case party => Claims(Seq(ClaimActAsParty(party)))
})
```

Or maybe you can make it a method on the companion object, whichever way you prefer. It should be quite easy to add a method that takes a map and uses it in a partial function to match this constructor.

Reply:
(also if you adopt this suggestion probably you can rename this to `AuthServiceStatic`)